File: crud-framework-core/src/main/java/com/antelopesystem/crudframework/crud/annotation/Deleteable.java
package com.antelopesystem.crudframework.crud.annotation;

import java.lang.annotation.*;

/**
 * Used to define a given entity as deleteable
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface Deleteable {

	/**
	 * Denotes whether the entity is soft deleteable or hard deleteable
	 */
	boolean softDelete();
}

File: crud-framework-core/src/main/java/com/antelopesystem/crudframework/crud/annotation/Deleteable.kt
package com.antelopesystem.crudframework.crud.annotation

/**
 * Used to define a given entity as deleteable
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class Deleteable(
        /**
         * Denotes whether the entity is soft deleteable or hard deleteable
         */
        val softDelete: Boolean
)